app/models/candidate.rb

```diff
@@ -24,6 +24,13 @@ def set_defaults
     self.vote ||= false
   end
 
+  def specialties_link
+    ApplicationController.helpers.collection_link(specialties){|x|[x.name, Rails.application.routes.url_helpers.chambers_specialty_path(order, x.slug)]}
```

Comment:
OK, this makes the "collection_link" thing above make more sense, but I'm not sure that `ApplicationController` is the right place for the helper? This makes me think you want to think about having a `CandidatesPresenter` that you can pass to the view that has this kind of logic.

Reply:
Also, you might be able to clean this up a bit by just passing the attribute you want to use on the thing in the collection and the URL helper, e.g., call as:

```ruby
collection_link(specialties,label: :name,order: order, link: :chambers_specialty_path)
```

and the helper would look like:

```ruby
def collection_link(collection, label:, order:, url_helper:)
  collection.map do |item|
    link_to(x.public_send(label), Rails.application.routes.url_helpers.public_send(url_helper, order, item.slug))
  end.join(', ').html_safe
end
```

(You might not have to do the full `Rails.application.routes.url_helpers` here - I think they're defined at the controller level, so if you keep the helper in `ApplicationController` you should just be able to do `send(url_helper, order, item.slug)` - URL helpers are always mysterious to me though.)

Reply:
OK I will try this.